drivers/i2c/i2c_nrfx_twim_rtio.c

```diff
@@ -188,6 +207,21 @@ static void event_handler(nrfx_twim_evt_t const *p_event, void *p_context)
 	const struct i2c_nrfx_twim_rtio_config *config = dev->config;
 	struct i2c_nrfx_twim_rtio_data *data = dev->data;
 	int status = p_event->type == NRFX_TWIM_EVT_DONE ? 0 : -EIO;
+	int ret = 0;
+
+	if (p_event->xfer_desc.type == NRFX_TWIM_XFER_TX) {
+		ret = dmm_buffer_out_release(config->common.mem_reg,
```

Comment:
Could this result in a large memcpy occuring in the ISR context? Won't this potentially disrupt time critical tasks?

At what size is the cost of a context swap worth considering perhaps to avoid holding up other tasks due to a memcpy?

Unlike the blocking driver which swaps back and forth to the calling thread, this does not. It expects you to be able to chain interupts together to avoid the context swap cost and reduce latency/overhead. But if you are doing large memcpy's in the ISR this isn't that beneficial. @bjarki-andreasen any thoughts?